### DISCUSSION TITLE
Twitter Polls voting for non premium users

### DISCUSSION THREAD

**Discussion by jcswlondon** [uBlockOrigin/uAssets#20668] [Ideas]
On and off since installing uBlockOrigin a few weeks ago I have been able to vote in Twitter polls without being a a premium subscriber to Twitter.
I have not been able to find any reference to Twitter Polls in any uBlockOrigin documentation. I therefore don't know if it is just a happy coincidence that certain versions block Twitter from stopping me voting or if it is a feature designed in to uBlockOrigin that sometimes works, sometimes not. I have tried watching the uBlockOrigin logger when I am casting votes and they are taken away but I am not familar with the logger and don't know what to look for. I did log this query in the Reddit help but apart from some up-votes, no replies.
I feel I should suggest allowing voting in Twitter polls for non premium users because it is a feature that is already there on and off. I am hoping that someone can figure why it sometimes works, sometimes doesn't and then perhaps have this feature mad to work more often than it does, it seems, by accident.

I use Firefox Developer, currently Version 120.0b9.

This is some of the text I posted to reddit;

With the latest few versions of UBlock I found my voting in polls again not working.

On 9th Nov I tried rolling back versions of UBlock to see if any recent earlier versions still worked by letting me vote. Indeed, once I got back to Version 1.51.0 I could again vote in Twitter polls.

Update 9 Nov 2023 - I can confirm that version 1.53.0 of uBlockOrigin is again allowing voting on twitter polls. Twitter stopped non premium users voting in polls months ago. What you see is your vote counted which then gets taken away within about 30 seconds or so.

I can also confirm that regardless of whether the current uBlockOrigin is allowing votes, the votes it has allowed in the past do remain - Twitter does show the tick over my choice indicating how I voted. So, uBlockOrigin really does sometimes be allowing votes, not just blocking the appearance of them being registered.

Update 10 Nov 2023 - No change to my Firefox, no update from 1.53.0 of uBlockOrigin but voting in polls was rejected again. Logged out, cleared all cookies, logged in again but still the same.

I then tried pre-release 1.53rc0 - no voting in polls. Then tried pre-release 1.53.3rc1 and can vote in polls again.

Update 14th Nov - again I found I can not vote in Twitter polls. Versions 1.5.4 through to pre-release 1.53.5rc6 do not allow poll votes.

**Comment by iam-py-test** [comment]
uBlock Origin does not bypass paywalls. This is either a bug with X or another browser extension you have installed. Please try disabling uBlock Origin and seeing if you can still vote.
Anyway, this is the wrong place to ask questions pertaining to uBlock Origin's filterlists. Such questions would be better asked on Reddit (as you did) or [in the uAssets discussions](https://github.com/uBlockOrigin/uAssets/discussions). Be aware the uBo team are volunteers; they do not have time to answer every question, and it may take them a little while to respond.
You may be able to find [some filterlists to do so](https://filterlists.com/), but be aware adding additional filterlists risks breaking websites. Any issues you experience with said filterlists should be reported to their maintainers.
Thank you.

**Reply by jcswlondon** [reply]
I am not a premium X user. That is why I am not allowed to vote in X polls. Voting in X polls was taken away from non premium members months ago. It is definitely uBlockOrigin that is sometimes allowing me to vote. I can never vote without uBlockOrigin enabled. I can sometimes vote with uBlockOrigin enabled. Then, after a few days, X stop me voting again - I update uBlockOrigin and I might get my vote back, or if not, perhaps a pre-release of uBlockOrigin gives me my vote back, or I have to wait for another uBlockOrigin release.
The only filters I have installed are youtube.
I will look in the areas you mentioned, thanks. I just find this issue frustrating because I don't think it is an intended feature but it is a nice feature that seems to work incidentally sometimes.

**Reply by iam-py-test** [reply]
Would you mind (when you are able to vote): clicking on the uBo icon, clicking the chat icon, scroll down, expand the _Troubleshooting information_ and paste that information here?
Thanks

**Reply by jcswlondon** [reply]
I don't mind at all and will do that.

In fact, voting worked for me just now. However, I am now very confused. uBo version had not changed from earlier. I could not vote earlier today but just tried now and could vote. That should not be possible for non premium user.
I then disabled uBo and refreshed the page - I could still vote. I logged out, cleared cookies and can still vote.
I promise you though, my previous testing of this had my voting rejected until I updated to a specific uBo version.

````yaml
uBlock Origin: 1.53.5rc6
Firefox: 120
filterset (summary):
 network: 104525
 cosmetic: 43591
 scriptlet: 18803
 html: 1100
listset (total-discarded, last-updated):
 default:
  user-filters: 4-0, never
  ublock-filters: 36291-152, 3h.1m
  ublock-badware: 7498-9, 3h.1m
  ublock-privacy: 661-0, 3h.1m
  ublock-unbreak: 2194-1, 3h.1m
  ublock-quick-fixes: 183-7, 3h.1m
  easylist: 75304-657, 3h.1m
  easyprivacy: 32876-766, 3h.1m
  urlhaus-1: 11030-0, 11h.2m
  plowe-0: 3720-1, 9d.23h.25m
filterset (user): [array of 4 redacted]
trustedset:
 added: [array of 2 redacted]
switchRuleset:
 added: [array of 2 redacted]
userSettings: [none]
hiddenSettings: [none]
supportStats:
 allReadyAfter: 211 ms (selfie)
 maxAssetCacheWait: 78 ms
popupPanel:
 blocked: 8263
 network:
  twitter.com: 8262
  ads-twitter.com: 1

```

**Reply by iam-py-test** [reply]
> filterset (user): [array of 4 redacted]

What do you have in your _My filters_ tab?
I know you said
>  The only filters I have installed are youtube.

So doubt that is the cause.
Thanks

**Reply by jcswlondon** [reply]
My Filters
_youtube.com##+js(set, yt.config_.openPopupConfig.supportedPopups.adBlockMessageViewModel, false)
    youtube.com##+js(set, Object.prototype.adBlocksFound, 0)
    youtube.com##+js(set, ytplayer.config.args.raw_player_response.adPlacements, [])
    youtube.com##+js(set, Object.prototype.hasAllowedInstreamAd, true)_

However, for the past few days I have not seen voting in polls fail for me. The usual fail is that a vote seems to be accepted but goes back to unvoted in about 30 seconds.  I can find numerous news articles saying that from March 2023 voting in polls would be taken away from non premium users. However, I can not find any recent news and the twitter.com help info on this is confusing - it says you must be a premium user to create a poll but I can not find any mention there of having to be a premium user to vote in polls. It is possible that X have quietly changed the new polls rule to effect only the creation of polls.  
Fo the time being this has become a non issue. If I don't see poll vote fail as a non premium user, there is no fault finding for me to carry on with because there is no fault.

**Reply by iam-py-test** [reply]
Unsurprisingly, those filters are exactly what you said they were. Sorry for not trusting.
I'm still curious why this happens. Sounds like an X bug, but then why would it go away when uBo is disabled?
Thanks

**Reply by Yuki2718** [reply]
> My Filters
> youtube.com##+js(set, yt.config.openPopupConfig.supportedPopups.adBlockMessageViewModel, false)
> youtube.com##+js(set, Object.prototype.adBlocksFound, 0)
> youtube.com##+js(set, ytplayer.config.args.raw_player_response.adPlacements, [])
> youtube.com##+js(set, Object.prototype.hasAllowedInstreamAd, true)_

They are the rules I asked redditor to test in May and totally useless. https://twitter.com/gorhill/status/1714683479068160268
